exercises/mongo-shell/intermediate-read.md

```diff
@@ -0,0 +1,352 @@
+delete martian._id
+martian.boxOffice = [
+    {"country": "North America", "revenue": 228.4},
+    {"country": "Others", "revenue": 401.7}
+]
+db.movieDetails.insertOne(martian)
+```
+
+What did we just do?!
+Simply put:
+* we created a variable called `martian`
+* added a reference to the document titled `"The Martian"` within that variable
+* went on to add another field called `boxOffice` to that variable which turns out to be an **array of objects**
+* inserted that variable in our collection as a document of its own
+
+Now when you run:
+```javascript
+db.movieDetails.find({"title": "The Martian"}).pretty()
+```
+you'll see that 2 documents are returned. One's the original document and the other is the one we just created with an additional `boxOffice` field. 
+
+The point of this was to add an **array of objects** field to our collection. Now onto the exercise where we'll see how to leverage `$elemMatch` to query this field. 
+
+**Exercise 2** :computer: 
+
+Take 5 minutes :alarm_clock: to find the movies where the `boxOffice` revenue in "North America" was greater than 
+* $200m 
+* $400m
+
+How many of you ran similar queries?
+```javascript
+db.movieDetails.find({"boxOffice.country": "North America", 
+    "boxOffice.revenue": {"$gt": 200}}).pretty()
+    
+db.movieDetails.find({"boxOffice.country": "North America", 
+    "boxOffice.revenue": {"$gt": 400}}).pretty()
+```
+
+:arrow_right: Note that the first query runs correctly by returning the document with "The Martian" `boxOffice`, but the second query returns the same result even though it is :x:!!! 
+There is no movie in our collection which has a revenue greater than $400m in North America. 
+
+This happened because the query ran the following checks: 
+* is `boxOffice.country` equal to "North America" :white_check_mark:
+* is `boxOffice.revenue` greater than $400m :white_check_mark:
+
+Both checks were satisfied at **different array positions**!!!
+
+Here's where `$elemMatch` comes in. It ensures that array elements present **in the same position** are evalauted. 
+
+Try the following 2 queries: 
+```javascript
+db.movieDetails.find({"boxOffice": 
+    {"$elemMatch": 
+        {"country": "North America", "revenue": 
+            {"$gt": 200}}}}).pretty()
+            
+db.movieDetails.find({"boxOffice": 
+    {"$elemMatch": 
+        {"country": "North America", "revenue": 
+            {"$gt": 400}}}}).pretty()
+```
+
+See the difference?!
+
+---
+
+### The "$size" query operator
+
+As the name suggests, this operator deals with the length of an array. 
+
+**Exercise 3** :computer: 
+
+* Which movies were filmed in just one country?
+
+    **query:**
+
+    ```javascript
+    db.movieDetails.find({"countries": {"$size": 1}}, 
+        {"_id": 0, "title": 1, "countries": 1})
+    ```
+    
+* Similar to the previous query, find the maximum number of countries that a movie(s) was shot in. It'll take a little trial and error but code away! 
+
+---
```

Comment:
Adding a placeholder for the Activities at the end.  This will need to be edited for the intermediate use cases 

```suggestion
---

## Group Activities
During this time the class should split into :five: groups to complete each section.

**Lab 1: :alarm_clock: 10 minutes** 

Share with the rest of the class the kind of movies you like by writing 3 interesting queries against this database, at the same time limiting the resulting documents to fields that you find relevant. 

For example, movies that were part of both the 'Action' and 'Adventure' genres! :boom: 


**Lab 2: :alarm_clock: 15 minutes**
Using `some other dbs we have` break into small groups and explore the documents in this database.  Create a small 2-3 slide long presentation about the data.  

Some questions to consider while exploring:
- How is this data structured?
- Is there anything interesing to me about this data?
- Is the data malformed, incorrect or incomplete?
- What are the challenges you faced while exploring this data?

**Lab 3: :alarm_clock: 15 minutes**
Present your results from **Lab 2** and answer any questions your classmates may have.
```

Reply:
```suggestion
---
## Group Activities
During this time the class should split into :five: groups to complete each section.

**Activity 1: :alarm_clock: 15 minutes** 

Let's figure out some weekend plans! Get together with the rest of your group and look at movies from a genre(s) you all like. Pick a couple of highly rated movies that you haven't watched yet and let the rest of the class know that these will be on your list! Also, are there some medium or low rated movies from that genre(s) that you've already watched? Do you agree with the rating? Why?

**Activity 2: :alarm_clock: 15 minutes**

Explore the different datasets available on your workstation.  Be creative and try to find interesting things.  
Some questions to consider while exploring:
- How is this data structured?
- Is there anything interesting to me about this data?
- Is the data malformed, incorrect or incomplete?
- What are the challenges you faced while exploring this data?

**Presentation: :alarm_clock: 25 minutes**
Present your results from **Group Activities** and answer any questions your classmates may have.
```

Reply:
I'll go ahead and commit these changes.  I'm okay with the added Activity, I like that you kept the exploration of the other data in here.  

What I said in the other PR still stands... You're leaning on the `movies` db too much. 

Imagine if you were in class and you hated movies, but we only did anything with movie data... you'd get bored, especially if you knew there was other data sitting right there and you loved shipwrecks and the titanic 🤷‍♂ 

Variety is key for this class to be interesting